Dockerfile

```diff
@@ -1,40 +1,59 @@
-FROM debian:jessie
+FROM python:2-slim
+MAINTAINER Mozilla Telemetry <telemetry-analysis-service@mozilla.org>
 
 ENV PYTHONUNBUFFERED 1
 ENV PYTHONPATH /app/
 
-RUN set -x \
-    && apt-get update \
-    && apt-get install locales -y \
-    && locale-gen en_US.UTF-8
-
 EXPOSE 8000
 
-RUN adduser --uid 1000 --disabled-password --gecos '' --no-create-home webdev
-
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-        build-essential python python-dev python-pip libpq-dev \
-        postgresql-client gettext sqlite3 libffi-dev \
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-
+# add a non-privileged user for installing and running the application
+RUN mkdir /app && \
+    chown 10001:0 /app && \
+    chmod g+w /app && \
+    groupadd --gid 10001 app && \
+    useradd --uid 10001 --gid 0 --home /app app
```

Comment:
According to a developer for RedHat's OpenShift platform that uses Docker extensively some Docker hosters won't allow us to run a Docker image as root and won't trust our use of USER either since we could map the root user inside the container to whatever user we want and then the hoster is screwed. So they ignore USER and assign a user ID that is different from other projects hosted there to the running container. But since that user ID isn't matching a real UNIX account it ends up setting the group ID to 0.

In other words, if we chose to use a group ID of 0, we're making sure our image is as compatible as possible to various Docker hosting services. If we don't strive to make Dockerflow host agnostic then we can update this to create an own group for the app user.